src/names.c

```diff
@@ -537,7 +537,11 @@ SEXP apply_name_spec(SEXP name_spec, SEXP outer, SEXP inner, R_len_t n) {
   }
 
   if (outer == strings_empty || outer == NA_STRING) {
-    return inner;
+    if (inner == R_NilValue) {
+      return chrs_empty;
```

Comment:
Oh maybe this answers my question from before about special behavior of `chrs_empty`